src/App.js

```diff
@@ -2,29 +2,39 @@ import React from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
+//React Custom Hook
+const useSemiPersistentState = () => {
+  
+  const [todoList, setTodoList] = React.useState(
+    JSON.parse(localStorage.getItem('savedTodoList') || '[]') 
```

Comment:
You can also not to parse an array, just `savedTodoList`. The code will be
` JSON.parse(localStorage.getItem('savedTodoList')) || '[]'`